Sources/Core/Pointer.swift

```diff
@@ -0,0 +1,56 @@
+// Copyright © 2022 Max Desiatov <max@desiatov.com>.
+// All Rights Reserved.
+// SPDX-License-Identifier: BSD-3
+
+public protocol _Pointer {
+  typealias Distance = Int
+
+  associatedtype Pointee
+
+  var _rawValue: Builtin.RawPointer { get }
+
+  init(_ _rawValue: Builtin.RawPointer)
+}
+
+public extension _Pointer {
+  @_transparent
+  init(_ from: OpaquePointer) {
+    self.init(from._rawValue)
+  }
+
+  @_transparent
+  init?(_ from: OpaquePointer?) {
+    guard let unwrapped = from else { return nil }
+    self.init(unwrapped)
+  }
+
+  @_transparent
+  init?(bitPattern: Int) {
+    guard bitPattern != 0 else { return nil }
```

Comment:
Any reason to prefer `guard` here rather than `if bitPattern == 0 { return nil }`